apps/src/labs/projects/ProjectManager.ts

```diff
@@ -111,14 +113,20 @@ export default class ProjectManager {
       return this.getNoopResponseAndSendSaveNoopEvent();
     }
     this.sourceToSave = source;
-    if (!this.canSave(forceSave)) {
-      if (!this.saveQueued) {
-        this.enqueueSave();
-      }
+    this.enqueueSaveOrSave(forceSave);
+  }
+
+  async rename(name: string, forceSave = false) {
+    if (this.destroyed || !this.lastChannel) {
+      // If we have already been destroyed or the channel does not exist,
+      // don't attempt to rename.
       return this.getNoopResponseAndSendSaveNoopEvent();
-    } else {
-      this.saveHelper();
     }
+    if (!this.channelToSave) {
+      this.channelToSave = _.cloneDeep(this.lastChannel);
```

Comment:
We have used JSON serialize/deserialize in Music Lab for deep clones. Curious if we should settle on a standard across the codebase?

Reply:
Ah I didn't realize we were already doing that. I went with lodash because doing json serialize/deserialize felt hacky to me, but it is the recommended way on a quick google. Do you all have a strong opinion here? fyi @sanchitmalhotra126

Reply:
Don't have a strong opinion, besides using JSON serialization for the sake of consistency?

Reply:
Same.  It also means one less third party dependency.

Reply:
👍 will update

Reply:
I updated. I will note that with json serialization you have to re-specify the type with `as <type>` which also feels hacky to me. If we aren't too opposed to a third party dependency `deepCopy` is much cleaner. We can change those as a follow-up though.